firestore-bundle-server/functions/__tests__/functions.test.ts

```diff
@@ -0,0 +1,286 @@
+    const bundleName = "query-with-a-collection-and-multiple-where-conditions";
+    const url = extUrl(bundleName);
+    const { data: bundle } = await axios(url);
+
+    const [metadata, documentMetadata, document] =
+      extractObjectfromBuffer(bundle);
+
+    /*** check metadata */
+    expect(metadata.metadata.id).toEqual(bundleName);
+    expect(metadata.metadata.totalDocuments).toEqual(1);
+
+    /*** check document metadata */
+    expect(documentMetadata.namedQuery.name).toEqual("example");
+
+    /*** check document */
+    expect(document.documentMetadata.queries[0]).toEqual("example");
+  });
+
+  xit("successfully returns a bundle using a document", async () => {
+    const bundleName = "single-document";
+    const url = extUrl(bundleName);
+    const { data: bundle } = await axios(url);
+
+    const [metadata, documentMetadata, document] =
+      extractObjectfromBuffer(bundle);
+
+    /*** check metadata */
+    expect(metadata.metadata.id).toEqual(bundleName);
+    expect(metadata.metadata.totalDocuments).toEqual(1);
+
+    /*** check document metadata */
+    expect(documentMetadata.namedQuery.name).toEqual("example");
+
+    /*** check document */
+    expect(document.documentMetadata.queries[0]).toEqual("example");
+  });
+
+  xit("successfully returns a bundle using multiple documents", async () => {
+    const bundleName = "multiple-documents";
+    const url = extUrl(bundleName);
+    const { data: bundle } = await axios(url);
+
+    const [metadata, documentMetadata, document] =
+      extractObjectfromBuffer(bundle);
+
+    console.log(documentMetadata);
+
+    /*** check metadata */
+    expect(metadata.metadata.id).toEqual(bundleName);
+    expect(metadata.metadata.totalDocuments).toEqual(2);
+
+    /*** check document metadata */
+    /**TODO: not working */
+    expect(documentMetadata.namedQuery.name).toEqual("example");
+
+    /*** check document */
+    expect(document.documentMetadata.queries[0]).toEqual("example");
+  });
+
+  xit("successfully returns a bundle using params", async () => {
+    const bundleName = "single-param";
+    const url = extUrl(bundleName);
+    const { data: bundle } = await axios(url);
+
+    const [metadata, documentMetadata, document] =
+      extractObjectfromBuffer(bundle);
+
+    /*** check metadata */
+    expect(metadata.metadata.id).toEqual(bundleName);
+    expect(metadata.metadata.totalDocuments).toEqual(0);
+
+    /*** check document metadata */
+    // expect(documentMetadata.namedQuery.name).toEqual("example");
+
+    /*** check document */
+    //expect(document.documentMetadata.queries[0]).toEqual("example");
+  });
+
+  it("successfully returns a bundle using clientCache", async () => {
+    const bundleName = "with-client-cache";
+    const url = extUrl(bundleName);
+    const { data: bundle } = await axios(url);
+
+    const [metadata, documentMetadata, document] =
+      extractObjectfromBuffer(bundle);
+
+    console.log(documentMetadata);
```

Comment:
Got a number of console.logs in place still

Reply:
Updated.